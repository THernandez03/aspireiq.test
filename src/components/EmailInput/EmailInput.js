import React, { useReducer } from "react";

import { TYPE, ADD_EMAIL, REMOVE_EMAIL } from '../../constants/actionTypes';
import Label from "../Label";
import Dropdown from "../Dropdown";
import styles from "./EmailInput.module.scss";

const Reducer = (state, action) => {
  switch (action.type) {
    case TYPE: {
      return { ...state, input: action.payload };
    }
    case ADD_EMAIL: {
      const { emails } = state;
      emails.add(action.payload);
      return { input: '', emails };
    }
    case REMOVE_EMAIL: {
      const { emails } = state;
      emails.delete(action.payload);
      return { ...state, emails };
    }
  }
};

export const EmailInput = ({ placeholder }) => {
  const [state, dispatch] = useReducer(Reducer, {
    input: "",
    emails: new Set()
  });

  return (
    <div className={styles.wrapper}>
      <div className={styles.labels}>
        {[...state.emails].map((email) => (
          <Label
            key={email}
            className={styles.label}
            onIconClick={() => {
              dispatch({ type: REMOVE_EMAIL, payload: email });
            }}
          >
            {email}
          </Label>
        ))}
      </div>

      <input
        type="email"
        value={state.input}
        placeholder={placeholder}
        onChange={(event) => {
          dispatch({ type: TYPE, payload: event.target.value });
        }}
        onKeyDown={(event) => {
          const { key, target } = event;
          const { value } = target;

          event.stopPropagation();

          if (!value || state.emails.has(value)) return;

          if (key === "Enter" || key === "Tab") {
            dispatch({ type: ADD_EMAIL, payload: value });
          }
        }}
      />

      <Dropdown
        term={state.input}
        hiddenValues={state.emails}
        onItemClick={(event) => {
          dispatch({ type: ADD_EMAIL, payload: event.target.innerText });
        }}
      />
    </div>
  );
};
