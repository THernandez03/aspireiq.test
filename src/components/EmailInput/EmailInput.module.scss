.wrapper{
  width: max-content;
  padding: 5px;
  box-shadow: 1px 1px 10px 0px #d7d7d7;
  border-radius: 5px;
  display: flex;

  > .labels{
    display: flex;

    > .label{
      margin: 0 1px;
    }
  }

  > input{
    border: 0;
    padding: 10px;
    width: 100%;

    &:focus{
      outline: 0;
    }
  }
}
