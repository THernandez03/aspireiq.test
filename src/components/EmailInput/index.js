export { EmailInput as default } from "./EmailInput";
