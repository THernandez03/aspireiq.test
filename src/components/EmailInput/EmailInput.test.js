import React from "react";
import { render } from "@testing-library/react";

import { EmailInput } from "./EmailInput";

describe("<EmailInput/>", () => {
  it("should render a EmailInput", () => {
    const { container } = render(<EmailInput />);
    expect(container).toMatchSnapshot();
  });
});
