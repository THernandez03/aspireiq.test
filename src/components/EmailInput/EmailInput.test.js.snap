// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailInput/> should render a EmailInput 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="labels"
    />
    <input
      type="email"
      value=""
    />
  </div>
</div>
`;
